core/trino-main/src/test/java/io/trino/operator/output/TestPagePartitioner.java

```diff
@@ -191,6 +199,23 @@ public void testOutputForMultipleSimplePages(PartitioningMode partitioningMode)
         assertThat(partitioned).containsExactlyInAnyOrderElementsOf(expected); // order is different due to 2 partitions joined
     }
 
+    @Test(dataProvider = "typesWithPartitioningModeAndNullBlock")
```

Comment:
Think about other edge cases too:
1. Empty block
2. Rle followed by empty block
3. Empty rle followed by null block
Can be separate PR too

Reply:
Since build must reset the appenders state I added these cases in one go in `io.trino.operator.output.TestPositionsAppender#testConsecutiveBuilds`